.github/workflows/ci.yml

```diff
@@ -56,8 +56,8 @@ jobs:
           docker build --label "org.label-schema.vcs-ref=$GITHUB_SHA" -t $DOCKER_IMAGE -f ./$SERVICE/Dockerfile .
 
       - name: Docker push
-        if: github.ref == 'refs/heads/master'
         run: |
-          docker push $DOCKER_IMAGE
+          GIT_BRANCH=${GITHUB_BASE_REF:-${GITHUB_REF#refs/heads/}}
+          docker push $DOCKER_IMAGE:$GIT_BRANCH
```

Comment:
As discussed, the push will fail here, as the docker image ` $DOCKER_IMAGE:$GIT_BRANCH` does not exist locally.   
I would suggest this change instead, to use the expected docker tag on build:

```diff
       - name: Set env variables
         run: |
-          echo "DOCKER_IMAGE=${DOCKER_IMAGE_PREFIX}_$SERVICE" >> $GITHUB_ENV
+          REF=${GITHUB_REF#refs/*/}
+          if [ "$REF" == "master" ]; then
+            IMAGE_TAG=latest
+          else
+            IMAGE_TAG=$REF
+          fi
+          echo "DOCKER_IMAGE=${DOCKER_IMAGE_PREFIX}_$SERVICE:$IMAGE_TAG" >> $GITHUB_ENV
           echo "SERVICE=$SERVICE" >> $GITHUB_ENV
         env:
           SERVICE: ${{ matrix.service }}
@@ -56,7 +62,7 @@ jobs:
           docker build --label "org.label-schema.vcs-ref=$GITHUB_SHA" -t $DOCKER_IMAGE -f ./$SERVICE/Dockerfile .
 
       - name: Docker push
-        if: github.ref == 'refs/heads/master'
+        if: github.event_name == 'push' # Don't push the image on pull_request run
         run: |
           docker push $DOCKER_IMAGE

```

Reply:
why don't push on pull request? can't I have an open Pull request during development?